figueroa-javier/script.js

```diff
@@ -0,0 +1,227 @@
+var express=require("express");
+var app=express();
+var bodyParser=require('body-parser');
+
+var server= app.listen(8080,'127.0.0.1' ,function(){
+console.log("Example")
+})
+/*---------------------------------------------------*/
+
+var socios=new Array();
+var prestamos=new Array();
+var libros=new Array();
+
+const DIA_EN_MILISEG=1000*60*60*24;
+
+function getNewId(){
+return Math.random().toString(36).substr(2,9);
+	
+}
+function encontrarIndicePorId(coleccion,idABuscar){
+	function esElId(elemento){
+		return elemento.id==idABuscar;
+
+}
+}
+
+function Socio(nombre,id){
+	this.nombre=nombre;
+	this.id=id;
+
+	
+}
+
+function Prestamo(id,id_libro,titulo_libro,nombre_socio,dias)
+{
+	this.id=id;
+	this.id_libro=id_libro;
+	this.titulo_libro=titulo_libro;
+	this.nombre_socio=nombre_socio;
+	this.fechaVencimiento=Date.now()+DIA_EN_MILISEG*dias
+}
+function Libro(titulo_libro,cantidad,id_libro){
+	this.titulo_libro=titulo_libro;
+	this.cantidad=cantidad;
+	this.id_libro=id_libro;
+	this.disponibles=function (prestamos){
+		var count=0;
+		for(const prest of prestamos){
+			if(prest.id_libro==id){
+				count++;
+			}
+		}
+		return this.cantidad-count;
+}
+}
+function Eliminarlibro(id){
+	for(var i=0;i<libros.length;i++){
+		console.log(libros[i]);
+		if(id==libros[i].id_libro){
+			libros.splice(i,1);
+			console.log("LIBRO ENCONTRADO");
+			return true;
+		}
+	}
+	return false;
+}
+
+function EliminarSocio(id){
+	for(var i=0;i<socios.length;i++){
+		if(id==socios[i].id){
+			socios.splice(i,1)
+			console.log("SOCIO ELIMINADO");
+			return true;
+		}
+	}
+	return 0;
+}
+function Buscarlibro(id_lib){
+	for(var i=0;i<libros.length;i++){
+		if(id_lib==libros[i].id_libro){
+			console.log("LIBRO ENCONTRADO");
+			return true;
+			
+		}
+	}
+}
+function Actualizarlibro(id_lib,cant){
+	for(var i=0;i<libros.length;i++){
+		if(id_lib==libros[i].id_libro){
+			libros[i].cantidad=cant;
```

Comment:
Falta comprobar la condición de que la nueva cantidad no tiene que ser menor a la cantidad de libros ya prestados

Reply:
This can also be checked in the RQ